### PR TITLE
Nav fastly keys

### PR DESCRIPTION
**Pull request by marcgear**
## Description
Biggest changes here have been to make it possible to merge the surrogate-key header from shift-node-api requests together to return in the express response.

To do this we create a postRequestHook that gets passed through to the methods of api-client.js when we're rendering server side.  This hook function merges the headers from the API response with those already on the express request.

All entry points to api-client have been updated to support the use of this 'options' param that contains the hook, and other pages that make API requests for cachable content should now use this mechanism to ensure all the requests have the right keys 

## Related Issue
shiftcommerce/trendy-golf#131

### Checklist

- [x] Appropriate commenting on functions/components (JSDoc format).
- [x] All SHIFT library versions bumped and published (utilised in this PR).
- [x] Unit tests added/amended.
- [x] Integration tests added/amended.
- [x] Environment variables supplied (.env.example also updated)

### How has/can this been tested?

- [x] Serverside rendering
- [x] Clientside rendering

### Describe how to test the change(s) here
Start the dev server for trendy or reference site and make some requests like:
```curl -I -X GET http://trendy.local:3000/```

observe the 'surrogate-key' header contains keys for the menu, the static content/category/whatever
 
### Dependencies Review
n/a

### CODE REVIEW

File: packages/shift-next/src/actions/account-actions.js [hunk 1]
```diff
@@ -11,11 +11,11 @@ const accountRequest = {
   successActionType: types.SET_ACCOUNT
 }
 
-export function fetchAccountDetails (store) {
```
Comment (marcgear) [inline packages/shift-next/src/actions/account-actions.js]: this unused argument was never passed by anything anywhere

File: packages/shift-next/src/lib/api-client.js [hunk 2]
```diff
@@ -18,6 +18,12 @@ class ApiClient {
     this.client = axios.create({
       baseURL: Config.get().apiHostProxy
     })
+
+    if (options.postRequestHook) {
+      this.postRequestHook = options.postRequestHook
+    } else {
+      this.postRequestHook = () => {}
+    }
```
Comment (marcgear) [inline packages/shift-next/src/lib/api-client.js]: This is where the magic happens - any callable passed in here gets called after the response.  We use this to pass in something that mutates the express request (req) without api-client ever being aware of it

File: packages/shift-next/src/lib/set-surrogate-keys.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        const currentParts = currentKey.split(' ')
+        const responseParts = responseKey.split(' ')
+        const combinedKey = [ ...new Set(currentParts.concat(responseParts)) ].join(' ')
+        res.set('surrogate-key', combinedKey)
+      }
+    }
+  }
+}
+
+export default setSurrogateKeys
```
Comment (marcgear) [inline packages/shift-next/src/lib/set-surrogate-keys.js]: Need to add a doc block

File: packages/shift-next/src/pages/static-page.js [hunk 4]
```diff
@@ -46,13 +47,17 @@ class StaticPage extends Component {
...
       const request = StaticPage.pageRequest(id)
-      const response = await new ApiClient().read(request.endpoint, request.query, dispatch)
+      const postRequestHook = (response) => {
+        return setSurrogateKeys(response.headers, req, res)
+      }
+      const response = await new ApiClient({ postRequestHook }).read(request.endpoint, request.query, dispatch)
 
       return response.data
     } catch (error) {
+      console.log(error)
```
Comment (marcgear) [inline packages/shift-next/src/pages/static-page.js]:
```suggestion
```

File: packages/shift-next/src/lib/with-redux-store.js [hunk 5]
```diff
@@ -39,7 +40,11 @@ export default (App) => {
       appContext.reduxStore = reduxStore
 
       if (!reduxStore.getState().menu.loaded) {
-        await reduxStore.dispatch(readMenu(Config.get().menuRequest))
+        let options = {}
+        if (appContext.req) {
+          options.postRequestHook = response => { setSurrogateKeys(response.headers, appContext.req, appContext.res) }
+        }
+        await reduxStore.dispatch(readMenu(Config.get().menuRequest, options))
```
Comment (marcgear) [inline packages/shift-next/src/lib/with-redux-store.js]: This is where the menu request gets made, and the postRequestHook gets passed through to the api-client

File: packages/shift-next/test/lib/set-surrogate-keys.test.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+// Libs
+import setSurrogateKeys from '../../src/lib/set-surrogate-keys'
+import { jsxText } from '@babel/types';
```
Comment (cbarratt) [inline packages/shift-next/test/lib/set-surrogate-keys.test.js]: Unsure if this is used in here or serves any purpose for the spec?

**Comment by cbarratt** [comment]
Looks like it may be worth reverting the version bump and pulling master, lerna is trying to jump to a version that is older than is currently in master (granted wasn't published but now is!)

**Comment by marcgear** [comment]
> Looks like it may be worth reverting the version bump and pulling master, lerna is trying to jump to a version that is older than is currently in master (granted wasn't published but now is!)

will do

**Pull request merged**